client/src/app/gui/rtp_streams_plot.rs

```diff
@@ -46,145 +52,264 @@ impl RtpStreamsPlot {
+                    .radio(setting == self.settings_x_axis, setting.to_string())
+                    .clicked()
+                {
+                    self.settings_x_axis = setting;
+                    self.requires_reset = true;
+                }
+            });
+        });
+        ui.horizontal_wrapped(|ui| {
+            let streams = &self.streams.borrow().streams;
+            let ssrcs: Vec<_> = streams.keys().collect();
+
+            ui.label("Toggle streams: ");
+            ssrcs.iter().for_each(|&ssrc| {
+                let selected = self.streams_visibility.get_mut(ssrc).unwrap();
+                let resp = ui.checkbox(
+                    selected,
+                    streams.get(ssrc).unwrap().display_name.to_string(),
+                );
+                if resp.clicked() {
+                    self.requires_reset = true
+                }
+            });
+        });
+        ui.vertical(|ui| {
+            ui.add_space(10.0);
+        });
+    }
+
+    fn plot_ui(&mut self, ui: &mut Ui, should_refresh: bool) {
+        if should_refresh {
+            self.refresh_points();
+        }
+
+        let plot = Plot::new("rtp-plot")
+            .show_background(false)
+            .show_axes([true, false])
+            .label_formatter(|name, _value| name.to_string());
+
+        if self.requires_reset {
+            plot.reset().show(ui, |plot_ui| {
+                self.draw_points(plot_ui);
+            });
+        } else {
+            plot.show(ui, |plot_ui| {
+                self.draw_points(plot_ui);
+            });
+        }
+        self.requires_reset = false;
+    }
+
+    fn draw_points(&mut self, plot_ui: &mut PlotUi) {
+        for (x, y, on_hover, color, radius) in &self.points_data {
+            let point = Points::new([*x, *y])
+                .name(on_hover)
+                .color(*color)
+                .radius(*radius);
+            plot_ui.points(point);
+        }
+    }
+
+    fn refresh_points(&mut self) {
+        self.points_data.clear();
+        let streams = self.streams.borrow();
+        let mut points_xy: Vec<(f64, f64)> = Vec::new();
+
+        streams
+            .streams
+            .iter()
+            .enumerate()
+            .for_each(|(stream_ix, (ssrc, stream))| {
+                if Self::stream_is_hidden(&self.streams_visibility, ssrc) {
+                    return;
                 }
-                self.requires_reset = false
-            })
+
+                Self::build_stream_points(
+                    &streams,
+                    &mut points_xy,
+                    stream_ix,
+                    &stream.rtp_packets,
+                    stream.display_name.to_string(),
+                    self.settings_x_axis,
+                    &mut self.points_data,
+                );
+            });
+    }
+
+    fn stream_is_hidden(streams_visibility: &HashMap<u32, bool>, ssrc: &u32) -> bool {
+        !(*streams_visibility.get(ssrc).unwrap())
+    }
```

Comment:
I general, I think that creating private associated methods is kinda pointless when you can put it outside of the `impl` block. I don't see any benefits, on the other hand you have to call with the `Self::something` and they're harder to test, but maybe I'm missing something.

Reply:
Agree, moved outside of class all those methods